crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -68,11 +68,55 @@ pub trait MapEntities {
 ///     fn map_entity(&mut self, entity: Entity) -> Entity {
 ///         self.map.get(&entity).copied().unwrap_or(entity)
 ///     }
+///
+///     fn mappings(&self) -> impl Iterator<Item = (Entity, Entity)> {
+///         self.map.iter().map(|(&source, &target)| (source, target))
+///     }
 /// }
 /// ```
 pub trait EntityMapper {
     /// Map an entity to another entity
     fn map_entity(&mut self, entity: Entity) -> Entity;
+
+    /// Iterate over all entity to entity mappings.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_ecs::entity::{Entity, EntityMapper};
+    /// # fn example(mapper: impl EntityMapper) {
+    /// for (source, target) in mapper.mappings() {
+    ///     println!("Will map from {source} to {target}");
+    /// }
+    /// # }
+    /// ```
+    fn mappings(&self) -> impl Iterator<Item = (Entity, Entity)>;
+}
+
+/// An [object safe](https://doc.rust-lang.org/reference/items/traits.html#object-safety) version
+/// of [`EntityMapper`]. This trait is automatically implemented for type that implements `EntityMapper`.
+pub trait DynEntityMapper {
```

Comment:
This should be linked to from `EntityMapper`.

Reply:
Good point! Added.